### ISSUE TITLE
beforeUpdate (new in v2.2) doesn't seem to work

### ISSUE THREAD

**Issue by simzen85** [vuejs/vue-router#1168]
I'm using vue-router 2.2.1 and have the below code

function requireAuth (to, from, next) {
// something
}

routes: [
    { path: '/submit_shelf/:mode', name: 'SubmitShelf', component: SubmitShelf, beforeEnter: requireAuth, beforeUpdate: requireAuth, forceRefresh: true },
    }
  ]

requireAuth isn't triggered when the route is changed from /submit_shelf/A to /submit_shelf/B

Can someone help to check this ? Thanks

**Comment by posva** [comment]
Hi!
Please follow the [Issue Reporting Guidelines](https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#issue-reporting-guidelines) and provide a minimal JSFiddle or Github repository containing a set of reproducible steps that can lead to the behaviour you described. Make sure to boil down the problem as much as possible. Thanks!

**Comment by baby2011** [comment]
beforeRouteUpdate hook function is executed twice, but i just want to fetch the request once.

**Comment by posva** [comment]
Forgot about this!
Closing for inactivity.
@baby2011 If the route gets updated only once but the hook is fired twice, open a new issue with a repro
